detect_secrets/filters/allowlist.py

```diff
@@ -17,6 +19,16 @@ def is_line_allowlisted(filename: str, line: str) -> bool:
         if regex.search(line):
             return True
 
+    previous_line = context.previous_line
+    regexes = _get_allowlist_nextline_regexes()
+
+    if ext[1:] in _get_file_based_allowlist_nextline_regexes():
+        regexes = _get_file_based_allowlist_nextline_regexes()[ext[1:]]
+
+    for regex in regexes:
+        if regex.search(previous_line):
+            return True
```

Comment:
Can we extract this into a function? I'm thinking something like:

```python
def _get_regexes_for_file(filename: str) -> Tuple[Iterable[Pattern], Iterable[Pattern]]:
    yield ...
```

then we can do:

```python
for payload, regex in zip([line, previous_line], _get_regexes_for_file(filename)):
    if regex.search(payload):
        return True
```

or something that allows us to take advantage of mostly similar code.

Reply:
It's been a while since I last wrote python, so hopefully I implemented this along the lines of what you were thinking. (Aside: thank goodness there are tests to verify whether it's working 😂) Please let me know if there's anything I ought to change 🙂